### PR TITLE
add dataproc kerberos tf example

### PR DESCRIPTION
**Pull request by jaketf**
This adds an example terraform module for spinning up a kerberized data lake with cross realm trust.
R: @jhambleton 
CC: @roderickyao

### CODE REVIEW

File: examples/kerberized_data_lake/main.tf [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+    "${var.metastore_cluster}_principal",
+    "${var.analytics_cluster}_principal",
+    "trust_${var.analytics_cluster}_${var.kdc_cluster}_principal",
+    "trust_${var.metastore_cluster}_${var.kdc_cluster}_principal",
+    "trust_${var.metastore_cluster}_${var.analytics_cluster}_principal",
+  ]
+}
+
+# This is a script to stage mock encrypted principals on GCS.
+resource "null_resource" "encrypted_principals" {
```
Comment (mik-laj) [inline examples/kerberized_data_lake/main.tf:85]: Should we add destroyer?

File: examples/kerberized_data_lake/main.tf [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+      module.service_accounts.iam_emails["${var.metastore_cluster}-sa"],
+      module.service_accounts.iam_emails["${var.analytics_cluster}-sa"],
+    ])
+  ]
+  prevent_destroy = "false"
+}
+
+
+locals {
+  users = [
```
Comment (mik-laj) [inline examples/kerberized_data_lake/main.tf:261]: Should these be variables?

File: examples/kerberized_data_lake/main.tf (hunk 1 above)
Comment (jaketf) [reply examples/kerberized_data_lake/main.tf:85]: done.

**Review by jhambleton — CHANGES_REQUESTED** [review]
nice work jake, this looks great! reviewed with minor change requests.

File: examples/kerberized_data_lake/main.tf [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+    "jake",
+    "jarek",
+    "kamil",
+    "daniel",
+    "core-data-svc",
+  ]
+  tenants = [for t in var.tenants : "${t}-svc"]
```
[the remark this answers is not in the document]
Comment (jaketf) [reply examples/kerberized_data_lake/main.tf:268]: Hmm I have a different thought. Requiring the user to use `-svc` naming convention is a leaky abstraction.
I think we should have separate vars for users (human) and tenant (non-humans) the `-svc` suffix is assumed for non-human principals by our init actions (implementation detail that should be abstracted away by this  module).

This TF module can enforce this with this line.
I've updated variables to have a separate users input.

[the remark this answers is not in the document]
Comment (jhambleton) [reply examples/kerberized_data_lake/main.tf:261]: agree with this update which makes it much cleaner.

**Review by jhambleton — APPROVED** [review]
lgtm

**Review by mik-laj — APPROVED** [review]

**Comment by mik-laj** [comment]
🎉

File: license-templates/LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
+Copyright 2020 Google Inc.
```
Comment (mik-laj) [inline license-templates/LICENSE.txt:1]: This file is not used. Can you say more about it?

**Pull request merged**